src/owner.rs

```diff
@@ -0,0 +1,357 @@
+use {Model, User};
+use util::{RequestUtils, CargoResult, internal, ChainError, human};
+use db::Connection;
+use curl::http;
+use pg::rows::Row;
+use rustc_serialize::json;
+use util::errors::NotFound;
+use std::str;
+
+
+/// Unifies the notion of a User or a Team.
+pub enum Owner {
+    User(User),
+    Team(Team)
+}
+
+/// For now, just a Github Team. Can be upgraded to other teams
+/// later if desirable.
+pub struct Team {
+    /// Github annoyingly has some APIs talk about teams by name,
+    /// and others by some opaque id. I couldn't find any docs
+    /// suggesting these ids are stable, so let's just always
+    /// ask github for it. *shrug*
+    github_id: i32,
+    /// Unique table id
+    cargo_id: i32,
```

Comment:
Currently the model-like structs all have public fields, and the crates.io ID is just called `id`